lib/authz/interface.go

```diff
@@ -0,0 +1,20 @@
+package authz
+
+import (
+	"context"
+	"tcplb/lib/core"
+)
+
+// ForwardingAuthorizer is a generic forwarding authorization policy that
+// controls which clients are allowed to forward connections to which upstreams.
+//
+// Multiple goroutines may invoke methods on a ForwardingAuthorizer simultaneously.
+type ForwardingAuthorizer interface {
```

Comment:
great link, thank you.  i think interfaces are very important to decouple designs, but moving the interface definition to the consumer side rather than producer side is reasonable. different consumers may only want different subsets of the concrete functionality & not agree on a common interface anyway.

authz is a much better package name

Reply:
Yep. Go interfaces are one of the more unique aspects of the language. Because we can do "adhoc" interfaces we have much more ways to decouple, which pushes APIs to prefer concrete types.